### PR TITLE
feat(treesitter): add support for setting query depths

### PR DESCRIPTION
**Pull request by lewis6991**
Requires: https://github.com/tree-sitter/tree-sitter/pull/2140

### CODE REVIEW

**Comment by clason** [comment]
I think we'll need to guard this behind a feature flag since this won't be in a release anytime soon. 

(See https://github.com/neovim/neovim/pull/14930)

**Comment by lewis6991** [comment]
I initially only raised this so I could test ts-context. I'm happy for this to live here for a while.

I'm not sure a feature flag is the best idea since it heavily changes the semantics.

**Comment by clason** [comment]
upstream PR is merged now; now we just have to wait for a new release.

**Comment by clason** [comment]
Maybe add a todo note in both locations to remove this when the minimum required version of treesitter is bumped to 0.20.9.

(Or we can wait for that release and just bump with this PR?)

**Comment by lewis6991** [comment]
Good to go?

File: cmake/FindTreesitter.cmake [hunk 1]
```diff
@@ -7,3 +7,20 @@ mark_as_advanced(TREESITTER_LIBRARY TREESITTER_INCLUDE_DIR)
 add_library(treesitter INTERFACE)
 target_include_directories(treesitter SYSTEM BEFORE INTERFACE ${TREESITTER_INCLUDE_DIR})
 target_link_libraries(treesitter INTERFACE ${TREESITTER_LIBRARY})
+
+# TODO(lewis6991): remove when min TS version is 0.20.9
+list(APPEND CMAKE_REQUIRED_INCLUDES "${TREESITTER_INCLUDE_DIR}")
+list(APPEND CMAKE_REQUIRED_LIBRARIES "${TREESITTER_LIBRARY}")
```
Comment (dundargoc) [inline cmake/FindTreesitter.cmake:13]: You need to remove these entries after the check

**Review by dundargoc — CHANGES_REQUESTED** [review]

**Review by clason — APPROVED** [review]

**Pull request merged**